File: monotonic-array/Solution896.java
// https://leetcode.com/problems/monotonic-array/

class Solution896 {

    // T: O(n), S: O(1)
    public boolean isMonotonic(int[] nums) {
        boolean increasing = true;
        boolean decreasing = true;
        for (int i = 0; i < nums.length - 1; i++) {
            if (nums[i] > nums[i + 1]) {
                increasing = false;
            }
            if (nums[i] < nums[i + 1]) {
                decreasing = false;
            }
        }
        return increasing || decreasing;
    }

    public static void main(String[] args) {
        var sol = new Solution896();
        var nums1 = new int[] { 1, 2, 2, 3 };
        System.out.println(sol.isMonotonic(nums1)); // true
        var nums2 = new int[] { 6, 5, 4, 4 };
        System.out.println(sol.isMonotonic(nums2)); // true
        var nums3 = new int[] { 1, 3, 2 };
        System.out.println(sol.isMonotonic(nums3)); // false
    }

}

File: monotonic-array/solution896.py
# https://leetcode.com/problems/monotonic-array/

from typing import List


class Solution896:

    # T: O(n), S: O(1)
    def isMonotonic(self, nums: List[int]) -> bool:
        increasing = True
        decreasing = True
        for i in range(len(nums) - 1):
            if nums[i] > nums[i + 1]:
                increasing = False
            if nums[i] < nums[i + 1]:
                decreasing = False
        return increasing or decreasing


def main() -> None:
    sol = Solution896()
    nums1 = [1, 2, 2, 3]
    print(sol.isMonotonic(nums1))  # True
    nums2 = [6, 5, 4, 4]
    print(sol.isMonotonic(nums2))  # True
    nums3 = [1, 3, 2]
    print(sol.isMonotonic(nums3))  # False


if __name__ == "__main__":
    main()
